src/extensions/resolveSource.ts

```diff
@@ -41,6 +55,33 @@ export async function resolveSource(extensionName: string): Promise<string> {
   return source;
 }
 
+/**
+ * Checks for and displays updateWarnings that apply to the given start and end versions.
+ * @param registryEntry the registry entry to check for updateWarnings in
+ * @param startVersion the version that you are updating from.
+ * @param endVersion the version you are updating to
+ */
+export async function checkForUpdateWarnings(
+  registryEntry: RegistryEntry,
+  startVersion: string,
+  endVersion: string
+): Promise<void> {
+  if (registryEntry.updateWarnings) {
+    for (const targetRange in registryEntry.updateWarnings) {
+      if (semver.satisfies(endVersion, targetRange)) {
+        const updateWarnings = registryEntry.updateWarnings[targetRange];
+        for (const updateWarning of updateWarnings) {
+          if (semver.satisfies(startVersion, updateWarning.from)) {
+            await displayUpdateWarning(updateWarning);
+            break;
```

Comment:
I added these breaks on the assumption that only 1 warning will ever apply to a given set of start and end version. If this isn't true, we should remove these.